### DISCUSSION TITLE
I use a certain software plugin emulation of a certain hardware EQ a lot :)

### DISCUSSION THREAD

**Discussion by n3wkular** [shanteacontrols/OpenDeck#158] [Show and tell]
![image](https://user-images.githubusercontent.com/107003685/172594183-7bdc4ed4-4f95-4a2e-bbfb-c34d0507841a.png)

I won't name the products, because most of us already know what they are. I use this EQ on most channels or tracks at various stages of the recording and mixing process. Fiddling with the tiny little software GUI and trying to dial in what I want to hear by dragging a mouse up and down, and then doing that for 20 tracks ... ugh.

So, I built this controller. Whenever I select a track or channel in my DAW, if the EQ plugin is loaded there, this controller is immediately active. Sure, there is the "soft takeover" to deal with, but that's a small price to pay for having actual knobs that work just like the real hardware. I swear I can smell the virtual tubes glowing sometimes :grin:

It uses an Arduino Mega (overkill, but it's all I had that day). I like to say I have a hand-wired edition of an iconic piece of hardware. And believe it or not, daisy-chaining the ground and V+ like this doesn't seem to cause any issues at all. I've been using it for almost 2 years now.

![image](https://user-images.githubusercontent.com/107003685/172596262-88710b35-67bc-4897-b0da-16fe72204d57.png)

With all the extra firepower on the Mega board, I am planning to build a second controller in the single space below this one in the rack. Probably some fancy routing controller for my DAW's FX chains (that I currently use custom actions for in the software, but wouldn't it be easier to have real buttons instead of having to remember tens of keyboard shortcuts all the time?!)

I'm working on several other controllers as well. Thanks for a killer platform in OpenDeck! Respect.

**Comment by paradajz** [comment]
Looks really great. Mind if I put it on website gallery? Thanks for sharing it!

> Sure, there is the "soft takeover" to deal with, but that's a small price to pay for having actual knobs that work just like the real hardware

Have you considered using encoders? The firmware supports remote sync option for encoders, so if your DAW can send the current knob state once you click on specific channel, the value for encoder syncs with your DAW, so next encoder increment or decrement will send the correct value. This avoids the need to fiddle with knob until it matches the state in DAW.

**Reply by n3wkular** [reply]
> Have you considered using encoders?

Yea, great point. In fact, I made a version that used encoders. But after about a week, I was annoyed by the lack of visual indicators on the control interface. I found that to "see" the settings of all the controls on the plugin at a glance, I had to load the plugin UI, and that sort of defeated the purpose. (I know, I know: we're supposed to listen with our ears, not our eyes! But the workflow didn't feel right and I compromised with potentiometers and the "soft takeover".)

I think I need to spend some more time with the advanced features of OpenDeck; the display support, in particular. When my DAW sends back the control state of a knob, there's got to be a way to display it. Something like this generic controller I sketched up years ago:
![Twiddle Box v0 1 (Front)](https://user-images.githubusercontent.com/107003685/172618377-c5577ea8-3970-4efe-a65d-937c612334c0.png)

In this mock up, each of the encoders is mapped to a parameter on the API-550B plugin on the Drum Buss track, and the SSD1306 display above each encoder shows the current mapped param and value in the software. (There would need to be a daughterboard or daughterboards to accomplish this with OpenDeck?) 

Other controllers do it like this:
![image](https://user-images.githubusercontent.com/107003685/172621576-73ff056f-ae1c-4058-86b1-c63c896965f0.png)

But, then you don't have a display for what encoder is mapped to what param.

And others solve that problem like this:
![image](https://user-images.githubusercontent.com/107003685/172623102-dcdc0ae7-309f-4dab-a7af-ae4214a1ced0.png)

Like I said, I need to dive into the more advanced stuff.

> Mind if I put it on website gallery?

Sure thing. It's really simplistic compared to the other things in the gallery, but why not?

Cheers!

**Reply by paradajz** [reply]
That's a cool mockup.

OpenDeck currently supports only 1 display. Support for more is doable but not supported at the moment. And yes, it would be easier to design a custom PCB for that many displays to avoid the mess of wires.

**Reply by n3wkular** [reply]
Exactly ^^. And either some basic muxing would need to be implemented with another microcontroller to "drive" the multiple displays, or a hairy network of small OpenDeck boards and a noisy MIDI event maelstrom needs to rage. I actually drew up the PCB for the displays in Eagle at some point. Maybe I'll look for it and revisit. Cheers!

**Reply by 86crush** [reply]
@n3wkular  I’ve been researching this idea for a long time and many people who attempted (or had sites based around custom builds) became defunct and got shut down right around 2019 🤦🏻‍♂️ what a shame. I’d love to speak to you about building some custom rigs for my rack i.e the 1176 etc. I think this is a MUCH better option than spending/maintaining a piece of hardware (especially an old one) I just hope you’d still be able to get the parts 🤦🏻‍♂️

**Reply by n3wkular** [reply]
@86crush Sure, pick my brain. That won't take long 😉 Are you thinking about the more advanced mock-up later in the thread or just the simple controller at the top?

**Reply by 86crush** [reply]
Awesome. So, it also depends if the UAD plugs (in particular) can be MIDI assigned. (Fingers crossed) I didn't see the advanced version in the thread that you mentioned, only the Pultec style one. This was really well done and looked as simple and tactile as the original! It looks like on average these can be 2u rack size. I would setup a separate USB hub (if USB could power them alone) I would love to start with an 1176 that had enough controls to accommodate all the different versions (bluey/blacky etc) in the series. Then we could move on from there. Also, how much $ do you think this would cost to make?

**Reply by n3wkular** [reply]
>  if the UAD plugs (in particular) can be MIDI assigned

In my experience, this is more a function of the plugin host or DAW. I typically use Reaper and it works without issue. To get a feel for how your DAW works, try mapping some control knobs from a MIDI controller you already own to the target plugin. For example: a LaunchKey or similar. You might find out in the process that you don't like the way it works and decide to abort ... or not.

> It looks like on average these can be 2u rack size

Yea, I made mine 2U so I could match the Pultec control layout. But it would be trivial to condense it to 1U. It's totally up to you.

> I would setup a separate USB hub (if USB could power them alone)

A USB hub can power it easily. No need for a separate hub, though; the voltage and current requirements for a small microcontroller are really modest.

> start with an 1176 that had enough controls to accommodate all the different versions (bluey/blacky etc)

Good plan. However, I would recommend against trying to replicate the RATIO buttons; it might make more sense to just use a simple potentiometer that maps to the various options (E.g., a 10K pot / 5 options = 2K "ranges"). Or maybe try a 5-way rotary switch. You'd have to experiment with that. Everything else requires simple potentiometers. (Assuming you don't want to replicate the VU meter and its settings ... and you DON'T want to do that; trust me.)

> how much $ do you think this would cost to make?

I will assume you are NOT asking what I would charge to make one.

That Pultec-style controller probably cost me less than $25 to make once I replaced the Arduino Mega. A cheap Black Pill for less than $5USD, potentiometers at about $0.60USD ea, knobs at less than $1USD ea, and just a few dollars in wire, solder, and a wooden project panel for the rack-mountable face. The invaluable part here is the OpenDeck platform itself; I still cannot believe it is open source!!

Do you have experience with Arduino or similar? If not, 90% of the effort will be figuring out exactly what to do with the other 10%. Of course, the knowledge you've gained when you finish is absolutely priceless.

**Reply by 86crush** [reply]
Very interesting. I will test all of that in Ableton (shouldn't be an issue) the entire reason why I researched this in the first place is that there is a guy who is selling a course on making your own arduino MIDI controllers (which is good) but at the same time very frustrating because he advertises a kit that you can buy on his website, that actually doesn't exist at all. You have to Frankenstein all these parts together on Amazon (most of which are not available due to supply chain, coming straight from China) so basically, it's "here's how to build something you may, or may not be able to get the parts for" so then I proceeded with the course and decided that I don't wanna solder, or build anything myself (much less machine/cut it) because of the hassle and would gladly pay someone to do this the right way. So, I am asking if you can build one and ship it to me. I'd expect to pay at least $100 (with shipping) is this possible for you to do? I can see needing at least six controllers if you can build them and provide support.

**Reply by n3wkular** [reply]
Ah, OK. I'm afraid I am not interested in making the controller, but I would gladly help you DIY it. I have my hands full with [clevelandmusicco.com](https://clevelandmusicco.com), and we do not plan to expand to MIDI controllers. For something like this, you should expect to pay someone significantly more than $100 to design, build, test, deliver, and support. Economy prices can only be realised at scale, so one-offs (or six-offs) must account for the cost of R&D, production setup, etc. amortised across just a few physical units.

So, by all means, if you fancy learning to solder, choose and source parts, and all that jazz, please don't hesitate to reach out again. I am always happy to share and learn alongside other makers!

**Reply by 86crush** [reply]
> Ah, OK. I'm afraid I am not interested in making the controller, but I would gladly help you DIY it. I have my hands full with [clevelandmusicco.com](https://clevelandmusicco.com), and we do not plan to expand to MIDI controllers. For something like this, you should expect to pay someone significantly more than $100 to design, build, test, deliver, and support. Economy prices can only be realised at scale, so one-offs (or six-offs) must account for the cost of R&D, production setup, etc. amortised across just a few physical units.
> 
> 
> 
> So, by all means, if you fancy learning to solder, choose and source parts, and all that jazz, please don't hesitate to reach out again. I am always happy to share and learn alongside other makers!

I appreciate the reply! And I completely understand what you mean. I think just sourcing the parts alone will cost me. Haha. May I ask where you sourced the parts? I think I will give it a go. Is there a blueprint for this? Having no experience with arduino, this may become a year-long project but I have noticed there is literally -no one- doing this (which blows my mind) I'm willing to go down the rabbit hole and start a work-bench! Again, I appreciate the thoughtful response!

**Reply by basst22778** [reply]
> Very interesting. I will test all of that in Ableton (shouldn't be an issue) the entire reason why I researched this in the first place is that there is a guy who is selling a course on making your own arduino MIDI controllers (which is good) but at the same time very frustrating because he advertises a kit that you can buy on his website, that actually doesn't exist at all. You have to Frankenstein all these parts together on Amazon (most of which are not available due to supply chain, coming straight from China) so basically, it's "here's how to build something you may, or may not be able to get the parts for" so then I proceeded with the course and decided that I don't wanna solder, or build anything myself (much less machine/cut it) because of the hassle and would gladly pay someone to do this the right way. So, I am asking if you can build one and ship it to me. I'd expect to pay at least $100 (with shipping) is this possible for you to do? I can see needing at least six controllers if you can build them and provide support.

I got all of my parts from AliExpress. 
Flashing the controller is the "hardest" part - but the documentation is very good.

With the blackpill you are restricted to 8 Potentiometers + 4 Encoders if you don't want to go deeper into electronics.
Other platforms support more Inputs - just see the documentation. https://github.com/shanteacontrols/OpenDeck/wiki/STM32F4-Black-Pill#potentiometers

A handwired Controller with 8 Potis takes 1-2hr to build.
If you live in the EU, I can send you a pre-flashed blackpill for a quick start. 

This was my "hello world"
![IMG_0083](https://user-images.githubusercontent.com/29714593/215597254-26350582-8be5-4149-97db-bbef714bd853.jpeg)

**Comment by n3wkular** [comment]
>I think just sourcing the parts alone will cost me.

It won't cost much, really. What you want to build is surprisingly simple. I'd start with something basic, like [a Black Pill](https://github.com/shanteacontrols/OpenDeck/wiki/STM32F4-Black-Pill). These can be purchased for less than $10USD. Just search eBay or the auction site of your choice for STM32F401CE (or other supported board). The OpenDeck wiki has all the answers you need, including which microcontrollers are supported.

>May I ask where you sourced the parts?

ebay. Tayda Electronics. The latter has wire, pots, several supported microcontroller boards, knobs, and more. The [wiki to the rescue](https://github.com/shanteacontrols/OpenDeck/wiki/Supported-components) again.

>Is there a blueprint for this?

Since you can build just about anything with OpenDeck, there is no blueprint. It isn't a project kit; it's a platform. It is limited only by your own creativity. I think what you are looking for is a schematic and a "pinout" table or diagram. Here's the [Black Pill pinout](https://github.com/shanteacontrols/OpenDeck/wiki/STM32F4-Black-Pill). It really is as simple as wiring components up to the correct pins and then using the web configurator to map controls as you like.

>I have noticed there is literally -no one- doing this (which blows my mind)

If you review this whole thread, there are a few examples of companies doing exactly this, but integrating MIDI CC-assignable knobs into [small keyboards](https://novationmusic.com/en/keys/launchkey-mini), or into [mixer stations](https://www.behringer.com/product.html?modelCode=P0B1X), for example. TC Electronic offers [dedicated hardware controllers](https://www.sweetwater.com/insync/first-look-tc-electronics-dvr-250dt/) for their flagship reverb and delay plugins; they do essentially what my neo-Pultec hardware does: maps physical controls to software parameters via MIDI CC.

Lastly, since all of this is new to you, I strongly recommend you start by getting a 10K potentiometer, some hookup wire, and a Black Pill. Then learn to ["flash" the board](https://github.com/shanteacontrols/OpenDeck/wiki/Flashing-STM32-boards). Then, wire up that single potentiometer, configure it with the [web configurator](https://config.shanteacontrols.com/#/), and use it to control a single parameter in your DAW. This is often called a "Hello, world" exercise, designed to introduce you to the mechanics of a platform. At that point, you will have a decent grasp on the process and the possibilities. Until then, there are too many fundamental unknowns for you to wonder about finer details. I promise you that once you make it this far, the rest will be many times clearer in your mind. And you'll be addicted to building controllers to interface with darn near everything in your studio 😁

**Reply by 86crush** [reply]
Amazing! I appreciate the information!

**Comment by paradajz** [comment]
Another option where you get to design your controller (layout only) and get the finished product is [Yaeltex](https://yaeltex.com/). Their controllers don't run OpenDeck, though, and you are somewhat limited with what components they offer and what their layout will be.
